presto-main/src/main/java/com/facebook/presto/execution/scheduler/ConsistentHashingNodeProvider.java

```diff
@@ -34,7 +33,6 @@
 import static java.util.Objects.requireNonNull;
 
 public class ConsistentHashingNodeProvider
-        implements NodeProvider
```

Comment:
why are we removing this ? so that we can plugin the nodeCount preference dynamically .. seems hacky

Reply:
I'm actually not sure if `NodeProvider` interface is necessary. This interface seems a little counterintuitive to me. A `NodeProvide` is passed to the split just to be called with a key. I wonder if a better interface would be to have something like `ConnectorSplit#getSoftAffinityNodeSelectionKey` that returns a String (or an object) similar to what is done for fragment level caching with `ConnectorSplit#getSplitIdentifier`. Then a scheduler can make a decision on it's own of how many preferred nodes are required. Thoughts?

Reply:
Another thing what this PR changes fundamentally, it moves the number of preferred nodes selection from a connector to scheduler. Previously it was a connector responsibility to request a certain number of preferred nodes. Now the scheduler will decide. 

From a practicality standpoint it feels like it's easier to make this selection on coordinator (a single property for every connector). But I wonder if there was some deeper thought process behind this that motivated to implement it this way.

@NikhilCollooru can you think of a situation when we think it would actually be preferred for a connector to chose of how many soft affinity nodes to chose for a certain split?

Reply:
yeah i dont see a usecase where you configure the node count based on connector. 
so moving the logic to scheduler seems the better way to do it.

Reply:
Sounds good. Let me follow up with a refactor PR.